.circleci/merge_base_branch.sh

```diff
@@ -0,0 +1,28 @@
+#!/usr/bin/env bash
+
+set -e
+
+# CircleCI doesn't give CIRCLE_PR_NUMBER in the environment for non-forked PRs. Wonderful.
+# In this case, we just need to do some ~shell magic~ to strip it out of the PULL_REQUEST URL.
+echo 'export CIRCLE_PR_NUMBER="${CIRCLE_PR_NUMBER:-${CIRCLE_PULL_REQUEST##*/}}"' >> $BASH_ENV
+source $BASH_ENV
+
+if [[ -n "${CIRCLE_PR_NUMBER}" ]]
+then
+    # Update PR refs for testing.
+    FETCH_REFS="${FETCH_REFS} +refs/pull/${CIRCLE_PR_NUMBER}/head:pr/${CIRCLE_PR_NUMBER}/head"
+    FETCH_REFS="${FETCH_REFS} +refs/pull/${CIRCLE_PR_NUMBER}/merge:pr/${CIRCLE_PR_NUMBER}/merge"
+
+    # Retrieve the refs
+    git fetch -u origin ${FETCH_REFS}
+
+    # Checkout PR merge ref.
+    git checkout -qf "pr/${CIRCLE_PR_NUMBER}/merge"
+
+    # Test for merge conflicts.
+    git branch --merged | grep "pr/${CIRCLE_PR_NUMBER}/head" > /dev/null
```

Comment:
So a comment to explain how this works is going to be necessary. I think this works by:

- The git checkout above will fail if there is no merge branch, and so we'll bail out due to `set -e`
- If it does exist, `git branch --merged` will list all branches that are merged into it. The grep will succeed if `pr/.../head` branch is merged into it, and fail if it isn't (and therefore bail out).
- If `pr/../head` is merged in then that means `pr/../merge` must be an up to date merge commit, and so we're done.